### DISCUSSION TITLE
Who wants to help? And with what?

### DISCUSSION THREAD

**Discussion by Explosion-Scratch** [Bijoujs/Bijou.js#30] [General]
If any of you guys have _any_ ideas for this project please comment! I'd like this project to continue expanding but I am at a loss for ideas! What else would make this better?

**Comment by ghost** [comment]
I wanna help, here's a snippet I have(for JS noobs)
```js
draw = function() {  
    background(220);  
      
    // Use color() function  
    let c = color('green');  
  
    // Use fill() function to fill color  
    fill(c);  
      
    // Draw a rectangle  
    rect(50, 50, 300, 200);  
}
```

**Reply by ghost** [reply]
> I wanna help, here's a snippet I have(for JS noobs)
> 
> ```js
> draw = function() {  
>     background(220);  
>       
>     // Use color() function  
>     let c = color('green');  
>   
>     // Use fill() function to fill color  
>     fill(c);  
>       
>     // Draw a rectangle  
>     rect(50, 50, 300, 200);  
> }
> ```

hold up that's p5

**Comment by ghost** [comment]
I don't know how to edit the snippets tho, help

**Comment by Explosion-Scratch** [comment]
> I wanna help, here's a snippet I have(for JS noobs)

Sorry but that's very specific and I frankly don't see many cases where this would be useful. If you can generalize this though that might be helpful!

**Comment by ghost** [comment]
> > I wanna help, here's a snippet I have(for JS noobs)
> 
> Sorry but that's very specific and I frankly don't see many cases where this would be useful. If you can generalize this though that might be helpful!

wdym?

**Comment by Explosion-Scratch** [comment]
> I don't know how to edit the snippets tho, help

In bijou.js add a new property to the end of the _temp object with the function, like this:
```js
let _temp = {
   //Some stuff here
   yourFunction: (string) => {
      console.log(string);
  }
}
``` 
Now you can use it like this:
```js
_$.yourFunction("Hello");
```

**Comment by ghost** [comment]
> > I wanna help, here's a snippet I have(for JS noobs)
> 
> Sorry but that's very specific and I frankly don't see many cases where this would be useful. If you can generalize this though that might be helpful!

Do you mean remove the values?

**Comment by Explosion-Scratch** [comment]
> > > I wanna help, here's a snippet I have(for JS noobs)
> > 
> > 
> > Sorry but that's very specific and I frankly don't see many cases where this would be useful. If you can generalize this though that might be helpful!
> 
> Do you mean remove the values?

🤔 Not really, what are several use cases of this function?

**Comment by ghost** [comment]
> > > > I wanna help, here's a snippet I have(for JS noobs)
> > > 
> > > 
> > > Sorry but that's very specific and I frankly don't see many cases where this would be useful. If you can generalize this though that might be helpful!
> > 
> > 
> > Do you mean remove the values?
> 
> 🤔 Not really, what are several use cases of this function?

Making animations or art in JS, like I do.

**Comment by ghost** [comment]
Also I barely know JS, I know a lot of HTML however.

**Comment by Explosion-Scratch** [comment]
whoops

**Comment by ghost** [comment]
lol

**Comment by ghost** [comment]
So if you need me, know i'm bad at JS but I can improve the site if you want.

**Comment by ghost** [comment]
>yep okay then



okay sir

**Comment by ghost** [comment]
hehe

**Reply by Explosion-Scratch** [reply]
Hehe what

**Reply by ghost** [reply]
I made it look like a quote

**Comment by ghost** [comment]
Is there anything I could help with, like the website?

**Comment by DabDatBass** [comment]
Haha, I really want to help with this. I have an account creation project ([accreate](https://github.com/DabDatBass/accreate)), maybe you can implement it?

**Reply by DabDatBass** [reply]
(npm and github release are broken, fixing them later)

**Comment by DabDatBass** [comment]
Want to help with functions altogether. I also have an npm account 👍

**Comment by ghost** [comment]
I'm really good with HTML and page building, if anything requires Python code I know that too

**Reply by ghost** [reply]
I also know like 0.0.1% of Haxe

**Reply by retronbv** [reply]
I don’t think there can be two decimals in a percent? 🤣

**Comment by xItzDiamondx** [comment]
I can help out in anyway

**Reply by Explosion-Scratch** [reply]
Sure thing! How do you want to help?

**Reply by xItzDiamondx** [reply]
I'd like to help develop.

**Reply by retronbv** [reply]
Anybody can develop! Just send pr’s

**Comment by xItzDiamondx** [comment]
Is there anything specific that I can do to help?

**Reply by retronbv** [reply]
read requests from issues or think up good ideas

**Comment by xItzDiamondx** [comment]
but can you give me an assignment to do

**Reply by retronbv** [reply]
one sec

**Reply by retronbv** [reply]
#82

**Reply by xItzDiamondx** [reply]
K

**Reply by Explosion-Scratch** [reply]
Wait, don't do that, I'm gonna remove that function altogether, instead just find new stuff to add!

**Comment by xItzDiamondx** [comment]
BTW, can I join the org

**Reply by retronbv** [reply]
inviting

**Reply by retronbv** [reply]
sent

**Reply by Explosion-Scratch** [reply]
that was ok retron but pls ask me next time

**Reply by retronbv** [reply]
You weren’t responding (offline)

**Comment by xItzDiamondx** [comment]
Is there anything that you think I should do specifically?

**Reply by Explosion-Scratch** [reply]
Hmm, I honestly don't know! You could go through the functions and make sure all of them have a working example in their JSDocs documentations. Testing the examples likely will reveal a lot of bugs.

**Reply by xItzDiamondx** [reply]
In which file?

**Reply by Explosion-Scratch** [reply]
https://github.com/Bijou-js/Bijou.js/blob/main/bijou.js <- That's the whole, compiled source code, but when you change something, make sure you change it in the appropriate file in /js file. Like this one:
![image](https://user-images.githubusercontent.com/61319150/114747016-fd088480-9d15-11eb-804f-5a7d475c9e99.png)
It says `@memberOf array` meaning that it's inside the `/js/array.js` file. So edit it there! Thanks for helping!

**Reply by imlens** [reply]
> https://github.com/Bijou-js/Bijou.js/blob/main/bijou.js <- That's the whole, compiled source code, but when you change something, make sure you change it in the appropriate file in /js file. Like this one:
> ![image](https://user-images.githubusercontent.com/61319150/114747016-fd088480-9d15-11eb-804f-5a7d475c9e99.png)
> It says `@memberOf array` meaning that it's inside the `/js/array.js` file. So edit it there! Thanks for helping!

How do I test it? (i'm a js noob but i do know js a little)

**Reply by Explosion-Scratch** [reply]
Just copy paste the function into the console (without `export`) and test it out in a variety of cases and stuff there. Then to contribute to the actual library just put your function at the top of the file you need and use jsdoc docs to document it

**Comment by imlens** [comment]
OK. Do I need to paste it into the console on a specific site? (sorry for making u explain too much)

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Reply by retronbv** [reply]
You don't :)

**Comment by imlens** [comment]
So I paste it into the console on any site.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Reply by retronbv** [reply]
Yes

**Comment by imlens** [comment]
OK. Thanks for explaining this to me.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Reply by retronbv** [reply]
Np
